src/calendar/view/eventeditor/CalendarEventEditDialog.ts

```diff
@@ -0,0 +1,235 @@
+		model,
+		recipientsSearch,
+		descriptionEditor,
+		startOfTheWeekOffset: getStartOfTheWeekOffsetForUser(locator.logins.getUserController().userSettingsGroupRoot),
+		timeFormat: getTimeFormatForUser(locator.logins.getUserController().userSettingsGroupRoot),
+		groupColors,
+	})
+		.addShortcut({
+			key: Keys.ESC,
+			exec: () => dialog.close(),
+			help: "close_alt",
+		})
+		.addShortcut({
+			key: Keys.S,
+			ctrl: true,
+			exec: () => okAction(assertNotNull(headerDom, "headerDom was null")),
+			help: "save_action",
+		})
+
+	if (client.isMobileDevice()) {
+		// Prevent focusing text field automatically on mobile. It opens keyboard and you don't see all details.
+		dialog.setFocusOnLoadFunction(noOp)
+	}
+	dialog.show()
+}
+
+/**
+ * show an edit dialog for an event that does not exist on the server yet (or anywhere else)
+ *
+ * will unconditionally send invites on save.
+ * @param model the calendar event model used to edit and save the event
+ */
+export async function showNewCalendarEventEditDialog(model: CalendarEventModel): Promise<void> {
+	let finished = false
+
+	const okAction: EditDialogOkHandler = async (posRect, model, finish) => {
+		if (finished) {
+			console.log("already finished, will not be saving new event")
+			return
+		}
+
+		if (model.editModels.whoModel.hasInsecurePasswords() && !(await Dialog.confirm("presharedPasswordNotStrongEnough_msg"))) {
+			console.log("not saving event: insecure passwords.")
+			return
+		}
+		try {
+			const result = await model.saveNewEvent()
+			if (result === EventSaveResult.Saved) {
+				finished = true
+				finish()
+			}
+		} catch (e) {
+			if (e instanceof UserError) {
+				// noinspection ES6MissingAwait
+				showUserError(e)
+			} else if (e instanceof BusinessFeatureRequiredError) {
+				model.canUseInvites = await showPlanUpgradeRequiredDialog(NewPaidPlans)
+			} else {
+				throw e
+			}
+		}
+	}
+	return showCalendarEventEditDialog(model, null, okAction)
+}
+
+/**
+ * show a dialog that allows to edit a calendar event that already exists.
+ *
+ * on save, will validate external passwords, account type and user intent before actually saving and sending updates/invites/cancellations.
+ *
+ * @param model the calendar event model used to edit & save the event
+ * @param identity the identity of the event to edit
+ * @param responseMail a mail containing an invite and/or update for this event in case we need to reply to the organizer
+ */
+export async function showExistingCalendarEventEditDialog(
+	model: CalendarEventModel,
+	identity: CalendarEventIdentity,
+	responseMail: Mail | null = null,
+): Promise<void> {
+	let finished = false
+
+	if (identity.uid == null) {
+		throw new ProgrammingError("tried to edit existing event without uid, this is impossible for certain edit operations.")
+	}
+
+	const okAction: EditDialogOkHandler = async (posRect, model, finish) => {
+		if (finished) {
+			return
+		}
+
+		if (
+			model.eventType === EventType.OWN &&
```

Comment:
can this login not be here but elsewhere?

Reply:
I'm not sure I understand, there's no login here

Reply:
I probably meant logic oops